pgoapi/poke_utils.py

```diff
@@ -1,11 +1,43 @@
 from __future__ import absolute_import
 import os
+from math import floor, sqrt
+from pgoapi.utilities import *
 from pgoapi.pokemon import Pokemon
 from pgoapi.game_master import PokemonData
 from pgoapi.protos.POGOProtos.Inventory import Item_pb2 as Enum_Items
 import csv
 
 
+#TCPM = CPM + ACPM
+#Stamina =  (BaseStamina + IndividualStamina) * TCPM
+#Attack = (BaseAttack + IndividualAttack) * TCPM
+#Defense = (BaseDefense + IndividualDefense) * TCPM
+#CP = MAX(10, FLOOR(Stamina0.5 * Attack * Def0.5 / 10))
+#ACPM = (Stamina/(BaseSta + IVSta)) - CPM
+def calcACPM(pokemon, pokemon_details):
+    if not all_in(['cp', 'cp_multiplier', 'individual_stamina', 'individual_attack', 'individual_defense'], pokemon):
+        return 0    
+
+    baseAttk = pokemon_details[str(pokemon['pokemon_id'])]['BaseAttack']
+    baseDef = pokemon_details[str(pokemon['pokemon_id'])]['BaseDefense']
+    baseStamina = pokemon_details[str(pokemon['pokemon_id'])]['BaseStamina']
+    cpm = pokemon['cp_multiplier']
+    return max(0, sqrt(sqrt( (100*pow(pokemon['cp'],2)) / ( pow((baseAttk + pokemon['individual_attack']), 2) * (baseDef + pokemon['individual_defense']) * (baseStamina + pokemon['individual_stamina'])  ) )) - cpm)
+
+def calcCP(pokemon, tcpm, pokemon_details):
```

Comment:
Hey are you referring the the MaxCP that I calculate in Web.py? The difference is that the one I made will take into acct evolutions and determine how much stardust/candy is needed. 

That MaxCP will show the max cp that pokemon can achieve. For example if the pokemon in question is a bulbasaur. That max cp will show the strongest that bulbasaur can get according to player level. Mine would show the strongest that bulbasaur can get if you evolved it to a Venasaur according to player level and how much dust and candy is needed.

What you show in the clip above is the acpm calculation. That is needed because acpm isn't included in the data dumps. It might be useful elsewhere if you also are not getting acpm from the servers. I don't know if that is so I didn't check but it is needed at least to calculate cp in web.py